src/builtins.c

```diff
@@ -1508,14 +1510,14 @@ size_t jl_static_show_x(JL_STREAM *out, jl_value_t *v, int depth)
             char *data = (char*)jl_data_ptr(v);
             n += jl_printf(out, "0x");
             for(int i=nb-1; i >= 0; --i)
-                n += jl_printf(out, "%02hhx", data[i]);
+                n += jl_printf(out, "%02" PRIx8, data[i]);
         }
         else {
             jl_value_t *fldval=NULL;
             JL_GC_PUSH1(&fldval);
             for (size_t i = 0; i < tlen; i++) {
                 if (!istuple) {
-                    n += jl_printf(out, ((jl_sym_t*)jl_svecref(t->name->names, i))->name);
+                    n += jl_printf(out, "%s", ((jl_sym_t*)jl_svecref(t->name->names, i))->name);
```

Comment:
You've tested it then with characters > 0x7f?  If the printf doesn't expect UTF-8, you might end up getting ANSI Latin1 characters, a common default for C char type.